core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -275,6 +276,11 @@ public HttpResponse doConfirmRename(@QueryParameter String newName) throws IOExc
      */
     @Restricted(NoExternalUse.class)
     public @Nonnull FormValidation doCheckNewName(@QueryParameter String newName) {
+
+        if (!isNameEditable()) {
```

Comment:
Maybe not needed? It is a dry-run, and it might still have use-cases when the name is not editable

Reply:
If I remove it, the Exception will be throws just some line after on https://github.com/jenkinsci/jenkins/blob/276d6899af202c288c9b5dabba05ed1d24cdfe20/core/src/main/java/hudson/model/AbstractItem.java#L264
I don't know how Exception are managed if it's not in FormValidation, I thought it was better for user experience. The renaming protection will still be there so I can remove this part.

Reply:
> Maybe not needed? It is a dry-run, 

so you get no error from a check but then the operation will always fail, this is wrong on so many levels.

> and it might still have use-cases when the name is not editable

those UC are wrong, this is checking if you can rename this to "foobar" - you can not if the thing is not editable - if someone somehow calls this they should be told "No this is not a valid name becausr the thing can not be renamed".